storage/client/client.go

```diff
@@ -1580,9 +1644,36 @@ func (c *StorageClient) RuntimeAccount(ctx context.Context, address staking.Addr
 		return nil, wrapError(err)
 	}
 
+	// Reconcile sdk balance(s) from node with the balance(s) fetched from the db.
+	c.upsertBalances(ch, &a)
+
 	return &a, nil
 }
 
+// Reads node sdk balances from ch and upserts them into acct.Balances, logging a
+// warning if the balances are mismatched.
+func (c *StorageClient) upsertBalances(ch chan *RuntimeSdkBalance, acct *RuntimeAccount) {
+	for nb := range ch {
+		found := false
+		// If the node balance differs from the balance in Nexus, defer to the node balance.
+		for i, b := range acct.Balances {
+			if b.TokenSymbol == nb.TokenSymbol {
+				found = true
+				if !b.Balance.Eq(nb.Balance) {
+					c.logger.Warn("sdk balance from node differed from dead-reckoned balance; defaulting to node balance", "address", acct.Address, "height", roothash.RoundLatest, "node_balance", nb.Balance.String(), "db_balance", b.Balance.String())
+					bPtr := &acct.Balances[i]
+					bPtr.Balance = nb.Balance
```

Comment:
ok I don't know how for...range loops work in go, but is it that `b.Balance = nb.Balance` doesn't really update acct?

Reply:
yep :\